### DISCUSSION TITLE
Is it possible to exclude response times/codes from parts of a scenario flow?

### DISCUSSION THREAD

**Discussion by EricMikaelKarlsson** [artilleryio/artillery#934] [Artillery]
Hello! 

I have a few steps in a flow that I need to perform for later parts of the scenario to function, such as login users in backend and other similar setup functions. I am however not interested in adding the results of these calls to the response time and response codes part of the scenario report. 

Is it possible to do this without spinning off these calls as separate js functions?

**Comment by hassy** [comment]
Hey @EricMikaelKarlsson 👋 It's not possible to *exclude* individual routes, but you can make Artillery report response times per URL/endpoint with this plugin: https://github.com/artilleryio/artillery-plugin-metrics-by-endpoint

Let me know if that works for your use-case!

**Reply by hassy** [reply]
btw as a bonus point, if your server reports `Server-Timing` (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server-Timing) Artillery will use that value to report request latency, i.e. network overhead won't be taken into account for a true measurement of your API's latency.

**Reply by EricMikaelKarlsson** [reply]
Thank you for the quick reply :).

This is an option I have tried, however the per-endpoint plugin does not appear to work with this plugin https://github.com/Nordstrom/artillery-plugin-cloudwatch (at least not out of the box) that is used on my site when running tests through AWS. It would appear that this plugin read all endpoints as one.

**Reply by hassy** [reply]
Ah indeed, it would appear the CloudWatch plugin does not process custom metrics (which is what the `metrics-by-endpoint` plugin relies on for reporting to external monitoring systems).

One solution would be to use custom JS and send metrics directly to CloudWatch yourself.

Another option would be to modify the CloudWatch plugin to take custom metrics into account when reporting to CloudWatch.

Sorry there's nothing that will just work out-of-the-box at the moment.

**Reply by EricMikaelKarlsson** [reply]
Thank you for looking into it. 
I will take a look at breaking these steps off into a separate "setup" js function as a first step. In my current use-case I do not need the response times so it does not matter if they are not logged as long as I can capture what they return for later parts in the scenario.
